File: app/src/main/java/com/example/vn008xw/reddit/ui/base/BasePresenter.java
package com.example.vn008xw.reddit.ui.base;

import android.support.annotation.VisibleForTesting;

import io.reactivex.disposables.CompositeDisposable;

public class BasePresenter<T extends BaseView> implements BasePresenterContract<T> {

    public CompositeDisposable mCompositeDisposable = new CompositeDisposable();

    private T mView;

    public T getView() {
        return mView;
    }

    @Override
    public void attachView(T view) {
        mView = view;
    }

    @Override
    public void removeView() {
        mCompositeDisposable.clear();
        mView = null;
    }
}

File: app/src/main/java/com/example/vn008xw/reddit/ui/base/BasePresenter.kt
package com.example.vn008xw.reddit.ui.base

import android.support.annotation.VisibleForTesting
import io.reactivex.disposables.CompositeDisposable

open class BasePresenter<T : BaseView> : BasePresenterContract<T> {

  @VisibleForTesting
  val mDisposable = CompositeDisposable()

  @VisibleForTesting
  var mView: T? = null

  override fun attachView(view: T) {
    mView = view
  }

  override fun removeView() {
    mDisposable.clear()
    mView = null
  }
}